### PR TITLE
SQL: Fixes issues with showing value column name prefix in legends 

### LINKED ISSUE #35390: Grafana 8.0 incorrectly adding prefix label to grouped data
**What happened**:
Updated to Grafana 8.0. I'm noticing all of my plots are adding as a prefix the variable name used. If a function is used (i.e. `COUNT()`, the function would be prefixed

**What you expected to happen**:
- Plot legend to display as expected

**How to reproduce it (as minimally and precisely as possible)**:
- Upgrade to Grafana 8.0
- Create a timeseries or  plot using psql

**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.0.0 (41f0542c1e)
- Data source type & version: PSQL 10
- OS Grafana is installed on: Linux
- User OS & Browser: OSX, Chrome
- Grafana plugins: None
- Others:
image

### PR DESCRIPTION
**Pull request by ryantxu**
Fixes #35390 (partially)

This is a SQL specific and back end only alternative to #35392

image

http://localhost:3000/d/JYola5qzz/datasource-tests-postgres?orgId=1&editPanel=4

### CODE REVIEW

File: pkg/tsdb/sqleng/sql_engine.go [hunk 1]
```diff
@@ -303,6 +303,20 @@ func (e *dataPlugin) executeQuery(query plugins.DataSubQuery, wg *sync.WaitGroup
 				errAppendDebug("failed to convert long to wide series when converting from dataframe", err, interpolatedQuery)
 				return
 			}
+
+			// Before 8x, a special metric label was used to name fields
```
Comment (torkelo) [inline pkg/tsdb/sqleng/sql_engine.go:307]:
```suggestion
			// Before 8x, a special metric column was used to name time series. The LongToWide transforms that into a metric label on the value column.
			// But that makes series name have both the value column name AND the metric name. So here we are removing the metric label here and moving it to the 
			// field name to get the same naming for the series as pre v8 
```

**Review by torkelo — APPROVED** [review]
This looks good to me

**Review by marefr — CHANGES_REQUESTED** [review]
Works in the case where you only have two fields, but not when you have more than one value fields with label metric. There's other cases when there's not a value and metric field as can be seen in [documentation](https://grafana.com/docs/grafana/latest/datasources/postgres/#time-series-queries):

The gdev dashboard has one such example for postgres unittest dashboard:
image

```sql
SELECT
  $__timeGroup("time_date_time",'5m'),
  min("value_double") as "min_value",
  max("value_double") as "max_value"
FROM test_data
WHERE $__timeFilter("time_date_time")
GROUP BY time
ORDER BY time
```

This gdev dashboard shows that it resolves one panel, but not the other two:

image

For example below query generates the following frame
```sql
SELECT
  $__timeGroup("createdAt",'$summarize'),
  max(value) as "value",
  hostname as "metric"
FROM 
  grafana_metric
WHERE
  $__timeFilter("createdAt") AND
  measurement = 'cpu' AND
  hostname IN($host)
GROUP BY time, metric
ORDER BY time
```

```json
{
  "response": {
    "results": {
      "A": {
        "frames": [
          {
            "schema": {
              "name": "results",
              "meta": {
                "executedQueryString": "SELECT\n  floor(extract(epoch from \"createdAt\")/60)*60 AS \"time\",\n  max(value) as \"value\",\n  hostname as \"metric\"\nFROM \n  grafana_metric\nWHERE\n  \"createdAt\" BETWEEN '2021-06-17T07:47:59.888Z' AND '2021-06-17T07:52:59.888Z' AND\n  measurement = 'cpu' AND\n  hostname IN('10.1.100.1','10.1.100.10','server1','server2')\nGROUP BY time, metric\nORDER BY time"
              },
              "fields": [
                {
                  "name": "time",
                  "type": "time",
                  "typeInfo": {
                    "frame": "time.Time"
                  }
                },
                {
                  "name": "value",
                  "type": "number",
                  "typeInfo": {
                    "frame": "float64",
                    "nullable": true
                  },
                  "labels": {
                    "metric": "server1"
                  }
                },
                {
                  "name": "value",
                  "type": "number",
                  "typeInfo": {
                    "frame": "float64",
                    "nullable": true
                  },
                  "labels": {
                    "metric": "server2"
                  }
                }
              ]
            },
            "data": {
              "values": [
                [
                  1623916080000,
                  1623916140000,
                  1623916200000,
                  1623916260000,
                  1623916320000
                ],
                [
                  101.94567116740645,
                  101.32983803008855,
                  98.24485134174023,
                  99.11082242003671,
                  98.20762076760217
                ],
                [
                  102.17495640876385,
                  101.49139611423894,
                  98.5697798472507,
                  99.34979998130595,
                  98.5910734483771
                ]
              ]
            }
          }
        ],
        "refId": "A"
      }
    }
  }
}
```

We need to decide whether wide data frames for SQL is the way forward. To me it is because of unified alerting and multi dimensions plus v8 is a good opportunity to do breaking change as long as we have a good workaround story. Let us know if you need help from backend platform figuring out this.

**Comment by hugohaggmark** [comment]
This solves this issue for me #35599

**Comment by torkelo** [comment]
@marefr 

SELECT
  $__timeGroup("time_date_time",'5m'),
  min("value_double") as "min_value",
  max("value_double") as "max_value"
FROM test_data
WHERE $__timeFilter("time_date_time")
GROUP BY time
ORDER BY time 

This query did not work in v7.5 for me , what format have selected time series or table?

We need to figure out somethign soon as v8 breaks almost all time series queries

**Comment by torkelo** [comment]
@marefr pushed an update that fixes it so that it works on all panels on that test dashboard, was a wrong condition that made the fix logic only apply to data frames with 2 fields

**Comment by marefr** [comment]
@torkelo 

> This query did not work in v7.5 for me, what format have selected time series or table? The query is missing a metric field as well so the above code should not have any effect ?

The old logic is, if there's no metric column selected, use the names of value columns as metric :)

**Comment by torkelo** [comment]
@marefr it's working now, the query below works well for me

```sql
SELECT
$__timeGroup("time_date_time",'5m'),
min("value_double") as "min_value",
max("value_double") as "max_value"
FROM test_data
WHERE $__timeFilter("time_date_time")
GROUP BY time
ORDER BY time
```

**Comment by torkelo** [comment]
pushed another update that refines the logic to only be applied when the orignal query returns 3 columns (so if you have more than one value field the fixing logic wont happen). 
Updated a integration test as well

**Review by marefr — APPROVED** [review]
LGTM

Works great :tada:

**Pull request merged**